src/main/java/link/infra/indium/renderer/render/TerrainBlockRenderInfo.java

```diff
@@ -16,14 +16,20 @@
 
 package link.infra.indium.renderer.render;
 
-import net.minecraft.block.Block;
+import me.jellysquid.mods.sodium.client.render.occlusion.BlockOcclusionCache;
 import net.minecraft.block.BlockState;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.Direction;
 
 public class TerrainBlockRenderInfo extends BlockRenderInfo {
 	private int cullCompletionFlags;
 	private int cullResultFlags;
+	protected final BlockOcclusionCache blockOcclusionCache;
+
+	public TerrainBlockRenderInfo(BlockOcclusionCache blockOcclusionCache)
+	{
+		this.blockOcclusionCache = blockOcclusionCache;
```

Comment:
Could you set this with a `setBlockOcclusionCache` method and reset it to null in `release`, to match `setBlockView`?